Functions/Administration/xASL_adm_DefineASLSequence.m

```diff
@@ -5,87 +5,86 @@
 % DESCRIPTION: This ExploreASL function tries to check what ASL sequence is
-% being processed, if this was not already defined in xQ.Sequence.
+% being processed, if this was not already defined in xQ.PulseSequenceType.
 % It does so by checking known combinations of readout dimensionality
-% (xQ.readoutDim) and Vendor, knowing the product sequences of the Vendors.
+% (xQ.MRAcquisitionType) and Vendor, knowing the product sequences of the Vendors.
 %
 % EXAMPLE: xQ = xASL_adm_DefineASLSequence(xQ);
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % Copyright 2015-2024 ExploreASL
 
 
-%% Check quantification fields readoutDim & xQ.Sequence
+%% Check quantification fields MRAcquisitionType & xQ.PulseSequenceType
 if nargin<2 || isempty(bVerbose)
     bVerbose = true;
 end
 
-if ~isfield(xQ, 'readoutDim') || isempty(xQ.readoutDim)
-    warning('xQ.readoutDim parameter missing');
+if ~isfield(xQ, 'MRAcquisitionType') || isempty(xQ.MRAcquisitionType)
+    warning('xQ.MRAcquisitionType parameter missing');
 end
 
 % Check for illegal sequence definitions
-if isfield(xQ,'Sequence')
-    if isempty(regexpi(xQ.Sequence, '(2d_epi|3d_grase|3d_spiral)'))
-        warning(['Unknown ASL sequence: ' xASL_num2str(xQ.Sequence)]);
+if isfield(xQ, 'PulseSequenceType')
+    if isempty(regexpi(xQ.PulseSequenceType, '(epi|grase|spiral)', 'once'))
+        warning(['Unknown ASL sequence: ' xASL_num2str(xQ.PulseSequenceType)]);
         fprintf('%s\n', 'Trying to fix this');
-        xQ = rmfield(xQ, 'Sequence');
+        xQ = rmfield(xQ, 'PulseSequenceType');
 
     end
 end
 
 %% Check vendor field
 % Assume vendor=GE for 3D spiral sequences 
-% (though this is tricky for special cases, Siemens and Philips are both
-% working on a 3D spiral)
-if isfield(xQ,'Sequence') && (~isfield(xQ, 'Vendor') || isempty(xQ.Vendor))
-    if strcmpi(xQ.Sequence,'3d_spiral')
+% (though this is tricky for special cases, Siemens and Philips are both working on a 3D spiral)
+if isfield(xQ, 'PulseSequenceType') && (~isfield(xQ, 'Vendor') || isempty(xQ.Vendor))
+    if strcmpi(xQ.PulseSequenceType,'spiral')
         warning('xQ.Vendor missing but 3D spiral sequence detected, assuming vendor GE');
         xQ.Vendor = 'GE';
     end
 end
 
 if ~isfield(xQ, 'Vendor') || isempty(xQ.Vendor)
     warning('xQ.Vendor missing, skipping determining ASL sequence');
-elseif  isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms|GE|Philips|Siemens'))
+elseif  isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms|GE|Philips|Siemens', 'once'))
     warning('Unknown Vendor specified in xQ.Vendor');
-elseif ~isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms'))
+elseif ~isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms', 'once'))
     fprintf('%s\n', 'Digital Reference Object ASL-DRO detected');
 end
 
 
 %% Try to work out which ASL sequence we have
```

Comment:
then this would become

%% Try to work out which ASL readout we have

(and similar perhaps throughout this function)

Reply:
Same as above - nothing has changed about the functionality.

Reply:
Not what I meant. I meant that we are checking the readout, not the sequence (sequence being labeling + readout)

Reply:
I will update to "readout".